api/src/main/resources/underlays/aou_synthetic.yaml

```diff
@@ -1338,6 +1331,34 @@ relationshipMappings:
         dataset: aou_synthetic
         table: concept_relationship
         column: concept_id_2
+  - name: condition_person
```

Comment:
My expectation was that for the general use case of the cohort builder, we would instead want to create a relationship between condition and condition_occurrence, and then apply two relationship filters ( **or** materialize the results of the condition search, and pass this back in as a literal to a condition_occurence filter ). This is because eventually modifiers will need to be applied at the condition_occurence row level. There are some more complex modifiers as well, including things like "at least X instances" or "occurring within X date".

Creating a direct relationship as you have here would simplify basic presence checks (i.e. does this participant have at least one row matching this condition), but I don't know if it's worth supporting this if we need to implement the general case anyways.

Reply:
I'm not sure a direct relationship between `condition` and `condition_occurrence` will produce the right cohort if there are search criteria defined in different domains. e.g.
- Define a cohort = people with covid who've had a mammogram (i.e. condition=covid and procedure=mammogram).
- Define a dataset on the cohort that includes all `condition_occurence` rows.

In this case, I think the `condition_occurrence` rows can't just be filtered on condition=covid, because then we'll include people with covid who haven't had a mammogram. We'd need to first get the set of people with `condition_occurrence` rows for covid and `procedure_occurrence` rows for mammogram. Then filter the `condition_occurrence` rows on that set of people. Does this sound right to you?

RE modifiers: I think we can still handle these. It means that when fetching rows of `condition_occurrence` for dataset builder, we'd use an `ArrayFilter` that contains the nested `RelationshipFilter` on `Person` & `Condition` and an e.g. `BinaryFilter` for checking the date.

Reply:
RE condition=covid example: right - here is logically how I imagine constructing your example dataset (just pseudo SQL - not meant to be representative of the best way of writing this):

```
SELECT * FROM condition_occurence WHERE person_id IN (my_cohort);
```

Defining my_cohort as:
```
SELECT person_id FROM person
INNER JOIN (condition_occurrence WHERE concept_id IN (covid concepts))
INNER JOIN (procedure_occurence WHERE concept_id IN (mammogram concepts))
```

(X concepts) could either be another nested filter / JOIN with the concepts table, or a materialized list of concept IDs, generated from a previous search.

RE modifiers: the X occurrences and other modifiers need to be filters applied to the `condition_occurence` entity, not the `condition` entity. Not sure if this was a typo or misunderstanding

This may be easier to talk through. Will setup some time tomorrow (feel free to move or cancel)

Reply:
Thanks for setting up the meeting. It will be very helpful to talk through this also.

After thinking about it some more, I think I'm following you better now. Here's a restatement, could you see if it makes sense to you?
- We need to be able to get a cohort of people with condition=covid.
  - One way to do this is to have a `people_condition` relationship and use a single `RelationshipFilter`. This is what I originally implemented.
  - Another way to do this is to have 2 relationships `people-condition_occurrence` and `condition_occurrence-condition` and use a nested `RelationshipFilter`. I pushed up a change to implement this with my latest commit.
  - Both of these will produce very similar queries when getting the list of people for a cohort.
- We also need to be able to get a dataset that includes condition occurrences where condition=diabetes for a cohort. 
  - To do this, we need two relationships `people-condition_occurrence` and `condition_occurrence-condition`. The former will be used for the filter that defines the cohort, and the later will be used for the filter that defines the concept set.
  - Since we're going to need these two separate relationships anyway, then for building a cohort, we can just navigate these same two relationships rather than defining a new one that effectively consolidates the two.
  
I pushed up a change now that I think implements this.